bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -667,7 +685,21 @@ private boolean updateRollerStatus(ShellySettingsStatus status, @Nullable Shelly
             getThing().postEvent(SHELLY_EVENT_ROLLER_CALIB, false);
         }
 
-        return updateChannels ? ShellyComponents.updateRoller((ShellyBaseHandler) getThing(), rs, cs.id) : false;
+        return updateChannels ? ShellyComponents.updateRoller((ShellyBaseHandler) getThing(), rs, rIdx) : false;
+    }
+
+    private int getRollerIdx(ShellyDeviceProfile profile, @Nullable Integer id) {
+        List<ShellySettingsRoller> rollers = profile.settings.rollers;
```

Comment:
Under which circumstances is `profile.settings.rollers` null?

Reply:
If the device is in relay, not in roller mode

Reply:
If the device is in relay, not in roller mode

Reply:
As far as I can see, `updateRollerStatus` is still called in the case (line 235)? Will `cs` be `null` in this case, or will we actually proceed to `getRollerIdx` and have `IllegalArgumentException` thrown?

Reply:
updateRollerStatus() as well as updateRelayStatus() will be caused on each status update. The check if the device is in roller or relays mode. If yea, status values will be processed. cs might be null, if cs.id is null it will be replaced by the id parameter.

getRollerIdx() is used, because the reported component id (roller.id) must not correlate to the index for settings.roller[}. Nevertheless, I need to have the case that the id is not found, therefore throwing an exception, which at the end results in thing getting OFFLINE with Communication Error.

Reply:
updateRollerStatus() checks for cover0 != null, which means not in roller mode / device doesn't support cover mode
updateRollerStatus() just returns if cover0 == null